File: asion-search/asion-search-core/src/test/java/org/asion/search/common/DataSourceTest.java
package org.asion.search.common;

import org.junit.Test;
import org.unitils.UnitilsJUnit4;
import org.unitils.database.annotations.TestDataSource;

import javax.sql.DataSource;

import java.sql.SQLException;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Asion.
 * @since 16/5/1.
 */
public class DataSourceTest extends UnitilsJUnit4 {


    @TestDataSource
    private DataSource defaultDataSource;

//    @TestDataSource("database2")
//    private DataSource dataSource;

    @Test
    public void testCheckDefaultDataSource() {
        assertThat(defaultDataSource).isNotNull();
        try {
            assertThat(defaultDataSource.getConnection()).isNotNull();
            System.out.println(defaultDataSource.getConnection());
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}

File: asion-account/asion-account-core/src/test/java/io/vincent/account/common/DataSourceTest.kt
package io.vincent.account.common

import org.junit.Assert
import org.junit.Test
import org.unitils.UnitilsJUnit4
import org.unitils.database.annotations.TestDataSource

import javax.sql.DataSource

/**
 * @author Asion.
 * @since 16/5/1.
 */
class DataSourceTest : UnitilsJUnit4() {


    @TestDataSource
    private val defaultDataSource: DataSource? = null

//    @TestDataSource("database2")
//    private val dataSource: DataSource? = null

    @Test
    fun testCheckDefaultDataSource() {
        Assert.assertNotNull(defaultDataSource)
        println(defaultDataSource!!.javaClass)
    }
}
